### PR TITLE
[api] Refactor /move API with additional validation checks

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Add extra validation checks for the /move API to catch edge case scenarios and improve error handling.

## How was this patch tested?

- Manually
- Unit tests

### CODE REVIEW

File: apps/filebrowser/src/filebrowser/api.py [hunk 1]
```diff
@@ -642,13 +642,48 @@ def rename(request):
...
+  if not request.fs.exists(source_path):
+    return HttpResponse('Source file or folder does not exist.', status=404)
+
+  # Check if the destination path is a directory
+  if not request.fs.isdir(destination_path):
+    return HttpResponse('Destination path must be a directory.', status=400)
+
+  # Check if destination path is parent of source path
+  if _is_destination_parent_of_source(request, source_path, destination_path):
+    return HttpResponse('Destination cannot be the parent directory of source.', status=400)
```
Comment (JohanAhlen) [inline apps/filebrowser/src/filebrowser/api.py:682]: Is this not the same as identical paths above? I take it it's different for folders but the message could be the same "Source and destination must be different"

**Review by JohanAhlen — APPROVED** [review]
LGTM

File: apps/filebrowser/src/filebrowser/api.py [hunk 2]
```diff
@@ -642,13 +642,48 @@ def rename(request):
...
+  # Check if paths are identical
+  if request.fs.normpath(source_path) == request.fs.normpath(destination_path):
+    return HttpResponse('Source and destination paths must be different.', status=400)
+
+  # Verify source path exists
+  if not request.fs.exists(source_path):
+    return HttpResponse('Source file or folder does not exist.', status=404)
+
+  # Check if the destination path is a directory
+  if not request.fs.isdir(destination_path):
```
Comment (tabraiz12) [inline apps/filebrowser/src/filebrowser/api.py:677]: does it reuse this for renaming as well like in unix. if are reusing this for renaming this might not work.

**Review by tabraiz12 — APPROVED** [review]

File: apps/filebrowser/src/filebrowser/api.py (hunk 2 above)
Comment (Harshg999) [reply apps/filebrowser/src/filebrowser/api.py:677]: No for rename operation we are not checking if destination path needs to be a directory or not.

File: apps/filebrowser/src/filebrowser/api.py (hunk 1 above)
Comment (Harshg999) [reply apps/filebrowser/src/filebrowser/api.py:682]: Somehow I feel from API perspective, the original error message makes sense here and clearly explains the issue. 

Assume you are hitting the API with source and destination path which are different but still you get the message that `Source and destination must be different` which feels vague and will require extra effort to figure out that the parent path cannot be same because that'd make move operation in same directory.

What do you think?
